plugins/coverimage.koplugin/main.lua

```diff
@@ -67,9 +71,39 @@ end
 
 function CoverImage:createCoverImage(doc_settings)
    if self.enabled and not doc_settings:readSetting("exclude_cover_image") == true then
-        local image = self.ui.document:getCoverPageImage()
-        if image then
+        local cover_image = self.ui.document:getCoverPageImage()
+        if cover_image then
+            local s_w, s_h = Device.screen:getWidth(), Device.screen:getHeight()
+            local i_w, i_h = cover_image:getWidth(), cover_image:getHeight()
+            local scale_factor = math.min(s_w / i_w, s_h / i_h)
+
+            if self.cover_image_background == "none" or scale_factor == 1 then
+                cover_image:writePNG(self.cover_image_path, false)
+                return
+            end
+
+            local scaled_w, scaled_h = i_w * scale_factor, i_h * scale_factor
+            local cover_image = RenderImage:scaleBlitBuffer(cover_image, scaled_w, scaled_h)
+
+            -- new buffer with screen dimensions,
+            local image = Blitbuffer.new( s_w, s_h, cover_image:getType() )
+            if self.cover_image_background == "black" then
+--                image:fill(Blitbuffer.COLOR_BLACK) -- not necessary, as BB are zero-initialized
```

Comment:
Please keep the `--` indented at the code level.
Also, no CI check seems to have run, but it will probably complain about the empty `then elseif`. When that's needed, one have to make it non empty with:
```lua
if something then
    do end -- luacheck: ignore 541
elseif
    other stuff
end
```
But I would probably leave the image.fill even if it's redundant. Or just chec for white and gray, and mention as a comment that black doesn't need it.

Reply:
Thanks for the hint. I will mention the behavior  on `Blitbuffer.new`.